acceptance/suites/tests/555-puppetserver-service/reboot.rb

```diff
@@ -0,0 +1,55 @@
+if (options[:type] == 'pe') then
+  pssn='pe-puppetserver'
+else
+  pssn='puppetserver'
+end
+
+test_name "Validates the ability of the #{pssn} service to reliably start after a reboot."
+
+max=3
+sleeptime=45
+retryCounter=0
+step "\n-----------\nValidate #{pssn} is up and running, then reboot, repeat #{max} times\n-----------"
+
+for i in 1..max 
+  step "\n-----\nLoop #{i}\n-----"
+   
+  step "check uptime"
+  origUptime=on(master, "awk '{print $1}' /proc/uptime").stdout.chomp.to_i
+
+  step "reboot"
+  #rebooting with on(master,'reboot') causes an IOError: stream end.  This seems like a sane workaround.
+  #QENG-1914 for more...
+  system "ssh -i #{options[:keyfile]} -o StrictHostKeyChecking=no root@#{master.reachable_name} \'reboot\'"
```

Comment:
One of the reasons why "max" started out set so high in this script is because I spent a lot of time testing methods similar to the above: nohup, background, sleep 2 && reboot, init 6, cron.  I didn't keep specific data, but I've previously rejected these methods as being intermittent, even before we try to move to multiple OSes.

The system -> ssh command works on my mac, and is already working in CI, at the end of the PE pipeline in qatests: http://jenkins-enterprise.delivery.puppetlabs.net/view/qatests/view/all/job/enterprise_qatests-puppet_integration-system_pe_021_puppet_service_3.8.x/LAYOUT=64mcd,PLATFORM=redhat6,UPGRADE_FROM=NONE,label=beaker/2/console

I would prefer to use the same patterns we always use with beaker, such as on(master, "somecommand"), but I spent a large amount of time trying to make that work in this case already.

Beyond reasons of good standard practice and concerns about the ssh command working in MacOS and in CI, is there a compelling reason why we should continue to struggle with the typical beaker method in this case that I'm missing?  Is there a way we can configure Ruby's Net::SSH so that it will gracefully accept that SSHD is shutting down?